sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -0,0 +1,398 @@
+                "Thus, the model cannot be loaded."
+            )
+        # Load model with the updated config
+        self.model, info = getattr(
+            transformers, prediction_model_class
+        ).from_pretrained(
+            self.model_path,
+            config=config,
+            output_loading_info=True,
+            ignore_mismatched_sizes=True,
+        )
+
+        # Freeze all loaded parameters
+        for param in self.model.parameters():
+            param.requires_grad = False
+
+        # Clamp all loaded parameters to avoid NaNs due to large values
+        for param in self.model.model.parameters():
+            param.clamp_(-1000, 1000)
+
+        # Reininit the weights of all layers that have mismatched sizes
+        for key, _, _ in info["mismatched_keys"]:
+            _model = self.model
+            for attr_name in key.split(".")[:-1]:
+                _model = getattr(_model, attr_name)
+            _model.weight = torch.nn.Parameter(
+                _model.weight.masked_fill(_model.weight.isnan(), 0.001),
+                requires_grad=True,
+            )
+
+        if self.validation_split is not None:
+            split = int(len(y) * (1 - self.validation_split))
+
+            train_dataset = PyTorchDataset(
+                y[:split],
+                config.context_length + max(config.lags_sequence),
+                X=X[:split] if X is not None else None,
+                fh=config.prediction_length,
+            )
+
+            eval_dataset = PyTorchDataset(
+                y[split:],
+                config.context_length + max(config.lags_sequence),
+                X=X[split:] if X is not None else None,
+                fh=config.prediction_length,
+            )
+        else:
+            train_dataset = PyTorchDataset(
+                y,
+                config.context_length + max(config.lags_sequence),
+                X=X if X is not None else None,
+                fh=config.prediction_length,
+            )
+
+            eval_dataset = None
+
+        training_args = deepcopy(self.training_args)
+        training_args["label_names"] = ["future_values"]
+        training_args = TrainingArguments(**training_args)
+
+        if self.fit_strategy == "minimal":
+            if len(info["mismatched_keys"]) == 0:
+                return  # No need to fit
+        elif self.fit_strategy == "full":
+            for param in self.model.parameters():
+                param.requires_grad = True
+        else:
+            raise ValueError("Unknown fit strategy")
+
+        trainer = Trainer(
+            model=self.model,
+            args=training_args,
+            train_dataset=train_dataset,
+            eval_dataset=eval_dataset,
+            compute_metrics=self._compute_metrics,
+            callbacks=self._callbacks,
+        )
+        trainer.train()
+
+    def _predict(self, fh, X=None):
+        if self.deterministic:
+            transformers.set_seed(42)
```

Comment:
Is `transformers` imported at this point?

Reply:
Yes since it is specified as a soft dependency of the Forecaster and if this soft-dependency exist then it is imported in line 21. And if the soft-dependency is not installed than it would raise an error before notifying the user that the soft-dependency is not installed.

Reply:
Sorry, I did miss those lines outside the estimator definition. I'm a bit concerned though about importing globally always if it's present. This means transformers and all its dependencies will be imported always even if I don't use anything other than ARIMA, right? It may have some unpleasant side effects, including higher load time etc.

Reply:
Not sure if I understand this correctly. 

If you are not importing this forecaster it should not import transformers or?